llama_cpp/llama_chat_format.py

```diff
@@ -1826,7 +2544,6 @@ def generate_schema_from_functions(functions, namespace="functions") -> str:
     def prepare_messages_for_inference(
         messages: List[llama_types.ChatCompletionRequestMessage],
```

Comment:
For very weird reason, the messages do not contain the `name` property when the role is `tool`.

Here, in the example I provided https://github.com/abetlen/llama-cpp-python/pull/1509#pullrequestreview-2215230004 the `messages` is equal to 

```json
[
....
   {
      "role":"tool",
      "content":"{\"location\": \"Hanoi, Vietnam\", \"temperature\": \"unknown\"}",
      "tool_call_id":"call_yQZk3sIQsIwGuco1lV4QVbTN"
   }
]
```

In the tool message we do not have the `name` which is required by the chat_format. leading to the error `'dict·object'·has·no·attribute·'name'` raised by jinja.